contracts/mixnet/src/storage.rs

```
@@ -48,6 +53,107 @@ pub fn config_read(storage: &dyn Storage) -> ReadonlySingleton<State> {
     singleton_read(storage, CONFIG_KEY)
 }
 
+fn total_mix_stake(storage: &dyn Storage) -> ReadonlySingleton<Uint128> {
+    singleton_read(storage, TOTAL_MIX_STAKE_KEY)
+}
+
+pub fn mut_total_mix_stake(storage: &mut dyn Storage) -> Singleton<Uint128> {
+    singleton(storage, TOTAL_MIX_STAKE_KEY)
+}
+
+fn inflation_pool(storage: &dyn Storage) -> ReadonlySingleton<Uint128> {
+    singleton_read(storage, INFLATION_POOL_PREFIX)
+}
+
+pub fn mut_inflation_pool(storage: &mut dyn Storage) -> Singleton<Uint128> {
+    singleton(storage, INFLATION_POOL_PREFIX)
+}
+
+fn total_gateway_stake(storage: &dyn Storage) -> ReadonlySingleton<Uint128> {
+    singleton_read(storage, TOTAL_GATEWAY_STAKE_KEY)
+}
+
+pub fn mut_total_gateway_stake(storage: &mut dyn Storage) -> Singleton<Uint128> {
+    singleton(storage, TOTAL_GATEWAY_STAKE_KEY)
+}
+
+pub fn total_mix_stake_value(storage: &dyn Storage) -> Uint128 {
+    match total_mix_stake(storage).load() {
+        Ok(value) => value,
+        Err(_e) => Uint128(0),
```

I'd argue that if we error out here, we should perhaps panic or something because isn't it the case that invalid `total_mix_stake` will mess up all rewarding?

---

Same comment applies to `total_gateway_stake_value` and `inflation_pool_value`